src/ngx_http_lua_kong_module.c

```diff
@@ -325,23 +325,31 @@ ngx_http_lua_kong_set_upstream_ssl(ngx_http_request_t *r, ngx_connection_t *c)
     STACK_OF(X509)              *chain;
     EVP_PKEY                    *pkey;
     X509                        *x509;
+    X509_STORE                  *store;
 #ifdef OPENSSL_IS_BORINGSSL
     size_t                       i;
 #else
     int                          i;
 #endif
 
     ctx = ngx_http_get_module_ctx(r, ngx_http_lua_kong_module);
-    if (ctx == NULL || ctx->upstream_client_certificate_chain == NULL) {
+
+    if (ctx == NULL || (ctx->upstream_client_certificate_chain == NULL &&
+                        ctx->upstream_trusted_store == NULL &&
+                        ctx->upstream_ssl_verify_depth == 0)) {
         ngx_log_debug0(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
-                       "skip overriding upstream SSL configuration, "
-                       "certificate not set");
+                    "skip overriding upstream SSL configuration, "
+                    "certificate and trusted store not set");
         return;
     }
 
     ngx_log_debug0(NGX_LOG_DEBUG_HTTP, c->log, 0,
                    "overriding upstream SSL configuration");
 
+    if (ctx->upstream_client_certificate_chain == NULL) {
+        goto skip_client_cert_key;
```

Comment:
We should avoid using `goto` if possible. A simple `if ctx->upstream_client_certificate_chain != NULL` should work well for that case.